File: Algorithms/452. Minimum Number of Arrows to Burst Balloons/python/solution.py
class Solution:
    def findMinArrowShots(self, points: list[list[int]]) -> int:
        if not points:
            return 0

        # sort by x_end
        points.sort(key=lambda x: x[1])

        arrows = 1
        first_end = points[0][1]
        for x_start, x_end in points:
            # if the current balloon starts after the end of another one,
            # one needs one more arrow
            if x_start > first_end:
                arrows += 1
                first_end = x_end

        return arrows

File: Algorithms/452. Minimum Number of Arrows to Burst Balloons/golang/solution.go
func findMinArrowShots(points [][]int) int {
	if len(points) == 0 {
		return 0
	}

	sort.Slice(points, func(i, j int) bool {
		return points[i][1] < points[j][1]
	})

	arrows, firstEnd := 1, points[0][1]

	for i := 0; i < len(points); i++ {
		if points[i][0] > firstEnd {
			arrows++
			firstEnd = points[i][1]
		}
	}

	return arrows
}
